src/convertToObject.js

```diff
@@ -2,17 +2,29 @@
 
 /**
  * Implement convertToObject function:
- *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
- *
+ * and values are the values of related CSS properties (see an exampl in
+ *[test file](./convertToObject.test.js))
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .filter(property => property.includes(':'))
+    .map(property => property.split(':'))
+    .reduce((prev, property) => {
```

Comment:
Good start. You have used 3 iterations here, but we can optimize this algorithm to 1 iteration. Let's do it.